File: DataStructures/Queue.cs
namespace DataStructures
{
    internal class Queuez<T>
    {
        static int MAX = 1000;
        int back;
        int front = 0;
        T[] queue = new T[MAX];

        public void PrintValues()
        {
            for (int i = front; i < back; i++)
            {
                Console.WriteLine(queue[i]);
            }
        }
        public void Enqueue(T val)
        {
            queue[back++] = val;
        }
        public T? Dequeue()
        {
            T first;
            if (back < front)
            {
                return default(T?);
            }
            else
            {
                first = queue[front];
                front++;
            }
            return first;
        }
        public void Count()
        {
            Console.WriteLine(back < front ? 0 : back - front);
        }
        public void Clear()
        {
            back = 0;
        }
        public bool Contains(T val)
        {
            for (int i = front; i < back; i++)
            {
                if (EqualityComparer<T>.Default.Equals(queue[i], val))
                {
                    return true;
                }
            }
            return false;
        }
        public T? Peek()
        {
            return back < front ? default(T) : queue[front];
        }
        public bool isEmpty()
        {
            return back <= front;
        }
    }
}

File: DataStructures/2. Queue/queue.ts
class Queuez<T> {
    private queue: Array<T> = []
    private front: number = 0
    private back: number = 0
    private MAX: number = 1000
    private count = 0;
    
  print() {
    console.log(this.queue);
  }
  enqueue(val:T) {
    if (this.count < this.MAX) {
        this.queue[this.back++] = val;
        this.count++
    }
    else {
        throw new Error("Out of Memory")
    }
  }
  dequeue() {
    if (this.count > 0) {
        this.count--;
        return this.queue.shift()
    }
    else
        return "Nothing to dequeue"
  }
  peek() {
    return this.count === 0 ? undefined : this.queue[this.front];
  }
  size() {
    return this.count;
  }
  isEmpty() {
    this.front = 0
    this.back == 0
    return this.count === 0;
  }
}
